compiler/damlc/lib/DA/Cli/Damlc/DependencyDb.hs

```diff
@@ -133,12 +136,40 @@ installDependencies projRoot opts sdkVer@(PackageSdkVersion thisSdkVer) pDeps pD
         ----------------------------
         forM_ dataDepsDars $ extractDar >=> installDar depsDir True
         forM_ dataDepsDalfs $ \fp -> BS.readFile fp >>= installDataDepDalf True depsDir fp
-        rdalfs <- getDalfsFromLedger dataDepsPkgIds
-        forM_ rdalfs $ \RemoteDalf {..} ->
+        exclPkgIds <- queryPkgIds Nothing depsDir
+        rdalfs <- getDalfsFromLedger dataDepsPkgIds exclPkgIds
+        forM_ rdalfs $ \RemoteDalf {..} -> do
             installDataDepDalf remoteDalfIsMain depsDir (remoteDalfName <.> "dalf") remoteDalfBs
+        -- Mark received packages as well as their transitiv dependencies as data dependencies.
```

Comment:
it adds the marker, but not for its dependencies. It's not necessary to traverse the dependencies and mark them as data dependencies as well to create a working package database. But I'd rather have a consistent view in the dependency database.

Reply:
I’m still confused, `getDalfsFromLedger` also returns dependencies doesn’t it? And we install all of what it returns via `installDataDepDalf`.

Reply:
`getDalfsFromLedger` only returns dependencies if we don't already have them present as normal dependencies.

Reply:
So the problem is the following:

1. data-dependency X depends on Dalf Y
2. Dalf Y is also present as a dependency
And this code flags Y is a data-dependency.

Why do we do this? E.g., if you consider Y = daml-stdlib, I don’t understand why you want to flag it is a data-dep.

Reply:
I added it only for consistency, so that the information that the depdendency is needed also as a data-dependency is noted. The info is not necessary to build up a working package database.

Reply:
But is the information actually correct? My understanding was that we prefer dependencies over data-dependencies so in the example above `daml-stdlib` would actually be pulled in as a dependency rather than a data-dependency.

Reply:
yes, when we construct the package db it will be pulled in as a dependency, not a data-dependency. I think we differ in the meaning of the data dependency marker. For me it marks a dependency that is contained in the transitive dependencies of a data-dependency, and I think for you it notes in what role it will be pulled in the package db.

Reply:
I guess it depends on how we use it. We have to resolve conflicts somehow if a dependency is resolved as both a data-dependency and a dependency. Given that we use it as a dependency later it seems more natural to me to resolve the marker the same way but I’m sure I’m missing something.

Reply:
My idea is that for packages that come from data-dependencies _and_ from dependencies, we can decide in the next step how to resolve them. While if we only mark them as dependency we need to take the decision already when building the dependency db. Also I think, currently with no caching they will also be marked as both.